query.go

```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
 	frame.Meta.ExecutedQueryString = query.RawSQL
 	frame.Meta.PreferredVisualization = data.VisTypeGraph
 
-	if query.Format == FormatOptionTable {
-		frame.Meta.PreferredVisualization = data.VisTypeTable
-		return data.Frames{frame}, nil
+	count, err := frame.RowLen()
+	if err != nil {
+		return nil, err
+	}
+	if count == 0 {
+		return nil, ErrorNoResults
 	}
 
-	if query.Format == FormatOptionLogs {
+	switch query.Format {
+	case FormatOptionTable:
+		frame.Meta.PreferredVisualization = data.VisTypeTable
+		return data.Frames{frame}, nil
+	case FormatOptionLogs:
 		frame.Meta.PreferredVisualization = data.VisTypeLogs
 		return data.Frames{frame}, nil
-	}
-
-	if query.Format == FormatOptionTrace {
+	case FormatOptionTrace:
 		frame.Meta.PreferredVisualization = data.VisTypeTrace
 		return data.Frames{frame}, nil
-	}
+	case FormatOptionMulti:
+		if frame.TimeSeriesSchema().Type == data.TimeSeriesTypeLong {
```

Comment:
I'm guessing the 3-plus-year-old deprecation warning on this method was... optimistic. (There are lots of other recent uses too.)

Reply:
Yeah, the dataplane repo has a new experimental implementation of timeseries that they're working on, but it's not there yet.